### DISCUSSION TITLE
Build 1.8.2 gives Make error at 12% with autoscan

### DISCUSSION THREAD

**Discussion by BobManMasher** [gerbera/gerbera#1714] [General]
Thought it was about time I compiled a new version of Gerbera so cloned 1.8.2 and used latest npupnp and followed the process I worked out last time for Pogo/Armel Debian Buster to get to the compile.

Cmake completed without errors and make -j4 gave the following error below at 12%. Can you give me some pointers please?

`[ 12%] Building CXX object CMakeFiles/libgerbera.dir/src/content/autoscan_list.cc.o
In file included from /usr/include/c++/8/map:61,
                 from /mnt/sata/gerbera/gerbera/src/config/config.h:28,
                 from /mnt/sata/gerbera/gerbera/src/util/thread_runner.h:36,
                 from /mnt/sata/gerbera/gerbera/src/util/timer.h:41,
                 from /mnt/sata/gerbera/gerbera/src/content/autoscan.h:40,
                 from /mnt/sata/gerbera/gerbera/src/content/autoscan.cc:32:
/usr/include/c++/8/bits/stl_map.h: In member function 'std::map<_Key, _Tp, _Compare, _Alloc>::mapped_type& std::map<_Key, _Tp, _Compare, _Alloc>::operator[](const key_type&) [with _Key = std::filesystem::__cxx11::path; _Tp = std::chrono::duration<long long int>; _Compare = std::less<std::filesystem::__cxx11::path>; _Alloc = std::allocator<std::pair<const std::filesystem::__cxx11::path, std::chrono::duration<long long int> > >]':
/usr/include/c++/8/bits/stl_map.h:499:8: note: parameter passing for argument of type 'std::_Rb_tree<std::filesystem::__cxx11::path, std::pair<const std::filesystem::__cxx11::path, std::chrono::duration<long long int> >, std::_Select1st<std::pair<const std::filesystem::__cxx11::path, std::chrono::duration<long long int> > >, std::less<std::filesystem::__cxx11::path>, std::allocator<std::pair<const std::filesystem::__cxx11::path, std::chrono::duration<long long int> > > >::const_iterator' {aka 'std::_Rb_tree_const_iterator<std::pair<const std::filesystem::__cxx11::path, std::chrono::duration<long long int> > >'} changed in GCC 7.1
    __i = _M_t._M_emplace_hint_unique(__i, std::piecewise_construct,
In file included from /usr/include/c++/8/map:60,
                 from /mnt/sata/gerbera/gerbera/src/config/config.h:28,
                 from /mnt/sata/gerbera/gerbera/src/util/thread_runner.h:36,
                 from /mnt/sata/gerbera/gerbera/src/util/timer.h:41,
                 from /mnt/sata/gerbera/gerbera/src/content/autoscan.h:40,
                 from /mnt/sata/gerbera/gerbera/src/content/autoscan.cc:32:
/usr/include/c++/8/bits/stl_tree.h: In member function 'std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::iterator std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::_M_emplace_hint_unique(std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::const_iterator, _Args&& ...) [with _Args = {const std::piecewise_construct_t&, std::tuple<const std::filesystem::__cxx11::path&>, std::tuple<>}; _Key = std::filesystem::__cxx11::path; _Val = std::pair<const std::filesystem::__cxx11::path, std::chrono::duration<long long int> >; _KeyOfValue = std::_Select1st<std::pair<const std::filesystem::__cxx11::path, std::chrono::duration<long long int> > >; _Compare = std::less<std::filesystem::__cxx11::path>; _Alloc = std::allocator<std::pair<const std::filesystem::__cxx11::path, std::chrono::duration<long long int> > >]':
/usr/include/c++/8/bits/stl_tree.h:2411:7: note: parameter passing for argument of type 'std::_Rb_tree<std::filesystem::__cxx11::path, std::pair<const std::filesystem::__cxx11::path, std::chrono::duration<long long int> >, std::_Select1st<std::pair<const std::filesystem::__cxx11::path, std::chrono::duration<long long int> > >, std::less<std::filesystem::__cxx11::path>, std::allocator<std::pair<const std::filesystem::__cxx11::path, std::chrono::duration<long long int> > > >::const_iterator' {aka 'std::_Rb_tree_const_iterator<std::pair<const std::filesystem::__cxx11::path, std::chrono::duration<long long int> > >'} changed in GCC 7.1
       _Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/8/bits/stl_tree.h: In member function 'std::pair<std::_Rb_tree_node_base*, std::_Rb_tree_node_base*> std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::_M_get_insert_hint_unique_pos(std::_Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::const_iterator, const key_type&) [with _Key = std::filesystem::__cxx11::path; _Val = std::pair<const std::filesystem::__cxx11::path, std::chrono::duration<long long int> >; _KeyOfValue = std::_Select1st<std::pair<const std::filesystem::__cxx11::path, std::chrono::duration<long long int> > >; _Compare = std::less<std::filesystem::__cxx11::path>; _Alloc = std::allocator<std::pair<const std::filesystem::__cxx11::path, std::chrono::duration<long long int> > >]':
/usr/include/c++/8/bits/stl_tree.h:2146:5: note: parameter passing for argument of type 'std::_Rb_tree<std::filesystem::__cxx11::path, std::pair<const std::filesystem::__cxx11::path, std::chrono::duration<long long int> >, std::_Select1st<std::pair<const std::filesystem::__cxx11::path, std::chrono::duration<long long int> > >, std::less<std::filesystem::__cxx11::path>, std::allocator<std::pair<const std::filesystem::__cxx11::path, std::chrono::duration<long long int> > > >::const_iterator' {aka 'std::_Rb_tree_const_iterator<std::pair<const std::filesystem::__cxx11::path, std::chrono::duration<long long int> > >'} changed in GCC 7.1
     _Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[ 13%] Building CXX object CMakeFiles/libgerbera.dir/src/content/autoscan_inotify.cc.o
`

**Comment by KarlStraussberger** [comment]
I suggest using the current HEAD which is close to the next release. Quite likely the it is fixed there already. Otherwise it's the place to fix it!

**Comment by KarlStraussberger** [comment]
Moved to issue #1715
